examples/rag_pipeline.py

```diff
@@ -0,0 +1,199 @@
+"""RAG pipeline"""
+
+import asyncio
+
+from pydantic import BaseModel
+
+from metagpt.const import DATA_PATH, EXAMPLE_DATA_PATH
+from metagpt.logs import logger
+from metagpt.rag.engines import SimpleEngine
+from metagpt.rag.schema import (
+    BM25RetrieverConfig,
+    ChromaIndexConfig,
+    ChromaRetrieverConfig,
+    FAISSRetrieverConfig,
+    LLMRankerConfig,
+)
+
+DOC_PATH = EXAMPLE_DATA_PATH / "rag/writer.txt"
+QUESTION = "What are key qualities to be a good writer?"
+
+TRAVEL_DOC_PATH = EXAMPLE_DATA_PATH / "rag/travel.txt"
+TRAVEL_QUESTION = "What does Bob like?"
+
+LLM_TIP = "If you not sure, just answer I don't know."
+
+
+class Player(BaseModel):
+    """To demonstrate rag add objs."""
+
+    name: str = ""
+    goal: str = "Win The 100-meter Sprint."
+    tool: str = "Red Bull Energy Drink."
+
+    def rag_key(self) -> str:
+        """For search"""
+        return self.goal
+
+
+class RAGExample:
+    """Show how to use RAG."""
+
+    def __init__(self):
+        self.engine = SimpleEngine.from_docs(
+            input_files=[DOC_PATH],
+            retriever_configs=[FAISSRetrieverConfig(), BM25RetrieverConfig()],
+            ranker_configs=[LLMRankerConfig()],
+        )
+
+    async def rag_pipeline(self, question=QUESTION, print_title=True):
+        """This example run rag pipeline, use faiss&bm25 retriever and llm ranker, will print something like:
+
+        Retrieve Result:
+        0. Productivi..., 10.0
+        1. I wrote cu..., 7.0
+        2. I highly r..., 5.0
+
+        Query Result:
+        Passion, adaptability, open-mindedness, creativity, discipline, and empathy are key qualities to be a good writer.
+        """
+        if print_title:
+            self._print_title("RAG Pipeline")
+
+        nodes = await self.engine.aretrieve(question)
+        self._print_result(nodes, state="Retrieve")
+
+        answer = await self.engine.aquery(question)
+        self._print_result(answer, state="Query")
+
+    async def rag_add_docs(self):
+        """This example show how to add docs, before add docs llm anwser I don't know, after add docs llm give the correct answer, will print something like:
```

Comment:
This example demonstrates how to add docs. 
Before adding docs the llm anwsers "I don't know". 
After adding docs, the llm provides the correct answer, will print something like: